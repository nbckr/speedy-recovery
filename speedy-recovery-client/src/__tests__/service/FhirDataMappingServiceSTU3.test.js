/*
* Speedy Recovery -- A patient-centred app based on the FHIR standard facilitating communication between paediatric
* patients, parents and hospital staff
*
* Copyright (C) 2019 University College London
*
* This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General
* Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option)
* any later version.
* This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied
* warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
* details.
* You should have received a copy of the GNU Affero General Public License along with this program. If not,
* see http://www.gnu.org/license/.
* */

/* This file tests the mapping from FHIR STU3 resources to our internal format*/

import {
  fhirMapAppointment,
  fhirMapCarePlan,
  fhirMapCondition,
  fhirMapMedication,
  fhirMapMedicationDispense,
  fhirMapPatient,
  fhirMapPractitioner
} from "../../service/FhirDataMappingService";
import fhirExamplePatient from "../test_input/fhir_resources_stu3/FhirExamplePatientSTU3.json";
import fhirExamplePatientMissingField from "../test_input/fhir_resources_stu3/FhirExamplePatientMissingFieldSTU3.json";
import fhirShortApp from "../test_input/fhir_resources_stu3/FhirSingleExampleAppointmentSTU3.json";
import fhirExampleCondition from "../test_input/fhir_resources_stu3/FhirExampleConditionSTU3.json";
import fhirExampleMedication from "../test_input/fhir_resources_stu3/FhirExampleMedicationSTU3.json";
import fhirExampleCarePlan from "../test_input/fhir_resources_stu3/FhirExampleCarePlanSTU3.json";
import fhirExampleMedicationDispense from "../test_input/fhir_resources_stu3/FhirExampleMedicationDispenseSTU3.json";
import fhirExamplePractitioner from "../test_input/fhir_resources_stu3/FhirExamplePractitionerSTU3.json";

test("map FHIR Patient resource to internal user data model", () => {
  const user = fhirMapPatient(fhirExamplePatient, "3");
  expect(user).toMatchSnapshot();
});

test("map FHIR Patient resource(with missing fields) to internal user data model", () => {
  const user = fhirMapPatient(fhirExamplePatientMissingField, "3");
  expect(user).toMatchSnapshot();
});

test("map FHIR Appointment resource to calendar data model", () => {
  const appointment = fhirMapAppointment(fhirShortApp, "3");
  expect(appointment).toMatchSnapshot();
});

test("map FHIR Condition resource to internal data model", () => {
  const condition = fhirMapCondition(fhirExampleCondition, "3");
  expect(condition).toMatchSnapshot();
});

test("map FHIR Medication resource to internal data model", () => {
  const medication = fhirMapMedication(fhirExampleMedication, "3");
  expect(medication).toMatchSnapshot();
});

test("map FHIR CarePlan resource to internal data model", () => {
  const carePlan = fhirMapCarePlan(fhirExampleCarePlan, "3");
  expect(carePlan).toMatchSnapshot();
});

test("map FHIR MedicationDispense resource to internal data model", () => {
  const medicationDispense = fhirMapMedicationDispense(
    fhirExampleMedicationDispense,
    "3"
  );
  expect(medicationDispense).toMatchSnapshot();
});

test("map FHIR Practitioner resource to internal data model", () => {
  const practitioner = fhirMapPractitioner(fhirExamplePractitioner, "3");
  expect(practitioner).toMatchSnapshot();
});
