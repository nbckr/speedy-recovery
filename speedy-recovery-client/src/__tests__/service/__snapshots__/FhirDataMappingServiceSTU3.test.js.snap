// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`map FHIR Appointment resource to calendar data model 1`] = `
Object {
  "appType": "Followup",
  "comment": "Further expand on the results of the MRI and determine the next actions that may be appropriate.",
  "created": 2013-10-10T00:00:00.000Z,
  "description": "Discussion on the results of your recent MRI",
  "end": 2013-12-10T11:00:00.000Z,
  "id": "example",
  "indication": "Severe burn of left ear",
  "location": "South Wing, second floor",
  "patient": "Peter James Chalmers",
  "practitioner": "Dr Adam Careful",
  "practitionerId": "example",
  "priority": 5,
  "start": 2013-12-10T09:00:00.000Z,
  "status": "booked",
  "supportingInfo": "DiagnosticReport/ultrasound",
  "title": "Brian MRI results discussion",
}
`;

exports[`map FHIR CarePlan resource to internal data model 1`] = `
Object {
  "activities": "1. Antenatal education; 2. Stress management; 3. Pregnancy diet education; ",
  "category": "Routine antenatal care",
  "id": "RES47187",
  "period": "from 1960-09-13 to 1961-04-11",
  "status": "completed",
}
`;

exports[`map FHIR Condition resource to internal data model 1`] = `
Object {
  "bodySite": "Left Ear",
  "clinicalStatus": "active",
  "onsetDateTime": 2012-05-24T00:00:00.000Z,
  "severity": "Severe",
  "summary": "Burnt Ear",
  "verificationStatus": "confirmed",
}
`;

exports[`map FHIR Medication resource to internal data model 1`] = `
Object {
  "content": "Vancomycin 500mg powder for infusion solution vial (product)",
  "form": "Injection Solution (qualifier value)",
  "id": "med0301",
  "imageURL": "Vancomycin Image",
  "isBrand": true,
  "isOverTheCounter": false,
  "name": "Vancomycin Hydrochloride (VANCOMYCIN HYDROCHLORIDE)",
  "producer": "Pfizer Laboratories Div Pfizer Inc",
}
`;

exports[`map FHIR MedicationDispense resource to internal data model 1`] = `
Object {
  "daysSupply": "30 days",
  "id": "smart-MedicationDispense-313",
  "name": "clopidogrel 75 MG Oral Tablet [Plavix]",
  "quantity": "30 tablets",
  "status": "completed",
  "whenHandedOver": "2009-07-08",
}
`;

exports[`map FHIR Patient resource to internal user data model 1`] = `
Object {
  "address": "534 Erewhon St, PleasantVille, Vic, 3999, US",
  "birthDate": "12/14/1963",
  "careProvider": "Practitioner/smart-Practitioner-71081332",
  "email": "Jim@example.org",
  "firstName": "Peter",
  "gender": "male",
  "id": "example",
  "name": "Peter James Chalmers",
  "phone": "(03) 5555 6473",
  "role": "Patient",
}
`;

exports[`map FHIR Patient resource(with missing fields) to internal user data model 1`] = `
Object {
  "address": "Unknown",
  "birthDate": "9/16/2011",
  "careProvider": "Practitioner/smart-Practitioner-71081332",
  "email": "Unknown",
  "firstName": "Steve",
  "gender": "male",
  "id": "smart-7777701",
  "name": "Steve Richey",
  "phone": "Unknown",
  "role": "Patient",
}
`;

exports[`map FHIR Practitioner resource to internal data model 1`] = `
Object {
  "birthDate": "1971-11-07",
  "gender": "female",
  "id": "f007",
  "name": "Simone Heps",
  "photo": "Unknown",
}
`;
